solend-sdk/src/classes/reserve.ts

```diff
@@ -29,70 +29,83 @@ export class SolendReserve {
   }
 
   private calculateSupplyAPY = (reserve: ParsedReserve) => {
-    const apr = this.calculateSupplyAPR(reserve);
-    const apy =
-      new BigNumber(1)
-        .plus(new BigNumber(apr).dividedBy(SLOTS_PER_YEAR))
-        .toNumber() **
-        SLOTS_PER_YEAR -
-      1;
-    return apy;
+    return calculateSupplyInterest(reserve, true). toNumber();
   };
 
   private calculateBorrowAPY = (reserve: ParsedReserve) => {
-    const apr = this.calculateBorrowAPR(reserve);
-    const apy =
-      new BigNumber(1)
-        .plus(new BigNumber(apr).dividedBy(SLOTS_PER_YEAR))
-        .toNumber() **
-        SLOTS_PER_YEAR -
-      1;
-    return apy;
+    return calculateBorrowInterest(reserve, true).toNumber();
   };
 
   private calculateSupplyAPR(reserve: ParsedReserve) {
-    const currentUtilization = this.calculateUtilizationRatio(reserve);
-
-    const borrowAPY = this.calculateBorrowAPR(reserve);
-    return currentUtilization * borrowAPY;
+    return calculateSupplyInterest(reserve, false). toNumber();
   }
 
   private calculateUtilizationRatio(reserve: ParsedReserve) {
     const totalBorrowsWads = new BigNumber(
       reserve.liquidity.borrowedAmountWads.toString()
-    ).div(WAD);
+    ).shiftedBy(-18);
+    const totalSupply = totalBorrowsWads.plus(
+        reserve.liquidity.availableAmount.toString()
+    );
     const currentUtilization = totalBorrowsWads
-      .dividedBy(
-        totalBorrowsWads.plus(reserve.liquidity.availableAmount.toString())
-      )
-      .toNumber();
+      .dividedBy(totalSupply);
 
     return currentUtilization;
   }
 
-  private calculateBorrowAPR(reserve: ParsedReserve) {
+  private calculateBorrowAPR = (reserve: ParsedReserve) => {
```

Comment:
i think this as caclulateUtilizationRatio are also in that other file

Reply:
i'd just edit it but i think i can't edit a fork

Reply:
Removed that method along with the caclulateUtilizationRatio from the reserve.ts file as they're no longer used.

Reply:
i think possible one might want to grab the utilization ratio but yeh seems fine now ill merge this

Reply:
actually kinda same with calculateBorrowAPR one probably would want it